Description: Fix FTBFS on mips, powerpc, s390x
 For some unknown reason, after right shifting, v is still greater than 256. It
 is possible cause by implementing right shift with rotation.
 .
 This patch just uses mask to fix v.
--- a/tsa2d32.cpp
+++ b/tsa2d32.cpp
@@ -627,6 +627,8 @@
       v = kk64 >> TSIN_HASH_SHIFT_64;
     }
 
+    v &= 0xff;
+
     if (v >= TSIN_HASH_N)
       p_err("error found %d", v);
 
